src/main.ts

```diff
@@ -251,8 +269,33 @@ export default class NewZettel extends Plugin {
     } else {
       titleContent = "";
     }
-    const fullContent = titleContent + content;
-    const file = await this.app.vault.create(path, fullContent);
+
+    let file = null;
+    const backlinkRegex = /{{link}}/g;
+    const titleRegex = /{{title}}/g;
+    if (this.settings.templateFile && this.settings.templateFile != "") {
+      let template_content = "";
+      try {
+        template_content = await this.app.vault.adapter.read(this.settings.templateFile);
+      } catch (err) {
+        new Notice(`Couldn't read template file ${this.settings.templateFile} make sure that it is a valid file...`);
+        return;
+      }
+
+      if (!titleRegex.test(template_content) || !backlinkRegex.test(template_content)) {
+        new Notice("Title {{title}} or Backlink {{link}} placeholder in template missing. Please add and try again...");
+        return;
+      }
+
+      const file_content = template_content.replace(titleRegex, titleContent).replace(backlinkRegex, fileLink);
+      file = await this.app.vault.create(path, file_content);
+      successCallback();
+    } else {
+      const fullContent = titleContent + fileLink;
+      file = await this.app.vault.create(path, fullContent);
```

Comment:
This is the previous file creation logic, that is used if no template is used.

Reply:
This is where the primary clober will be but not sure how the new stuff for the aliases will interface with your stuff if at all in the template version but if I remember correctly templater does not work correctly in the frontmatter. (crying emoji)